halo/evmstaking2/keeper/keeper.go

```diff
@@ -35,21 +32,23 @@ var (
 // Keeper also implements the evmenginetypes.EvmEventProcessor interface.
 type Keeper struct {
 	eventsTable     EVMEventTable
-	ethCl           ethclient.Client
+	ethCl           ethclient.EngineClient
```

Comment:
this module doesn't use any EngineAPI functions, so don't think we should use EngineClient.